### PR TITLE
Joes ansible.refactor update for logging and cronjob to clean old log files.

### PR DESCRIPTION
**Pull request by jpappa200**
## What does this PR (Pull Request) do?

- [x] This PR fixes #REPLACE_ME OR is not related to any Issue 

## Which Traffic Control components are affected by this PR?

- Traffic Router

## What is the best way to verify this PR?
These additions were run against the QA CDN and we got the 
expected results without issue.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: infrastructure/ansible/roles/traffic-router/defaults/main.yml [hunk 1]
```diff
@@ -171,3 +170,11 @@ tr_logging_opts:
...
   - 1catalina.org.apache.juli.AsyncFileHandler.rotatable = true
   - 1catalina.org.apache.juli.AsyncFileHandler.maxDays = 30
+
+# TR cronjobs to deploy
+tr_crontab:
+  # Cleanup log files more than 2 days old.
+  access_cleanup:
+    schedule: '10 */4 * * *'
+    user: root
+    job:  /bin/find /opt/traffic_router/var/log/access.log.* -mmin +$((60*48)) -exec rm -f {} \; > /dev/null 2>&1
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic-router/defaults/main.yml]: We should use {{tr_log_dir}} here instead of hard coding the path.

**Review by jhg03a — CHANGES_REQUESTED** [review]

File: infrastructure/ansible/roles/traffic-router/templates/crontab.j2 [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# File deployed via ansible push
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic-router/templates/crontab.j2]: The license header I believe has to go first here.

**Review by jhg03a — CHANGES_REQUESTED** [review]

File: infrastructure/ansible/roles/traffic-router/templates/crontab.j2 (hunk 2 above)
Comment (jhg03a) [reply infrastructure/ansible/roles/traffic-router/templates/crontab.j2]: Double check one of the other places we do this.  I could be wrong on this.

File: infrastructure/ansible/roles/traffic-router/defaults/main.yml (hunk 1 above)
Comment (jpappa200) [reply infrastructure/ansible/roles/traffic-router/defaults/main.yml]: updated to use {{tr_log_dir}}

File: infrastructure/ansible/roles/traffic-router/templates/crontab.j2 (hunk 2 above)
Comment (jpappa200) [reply infrastructure/ansible/roles/traffic-router/templates/crontab.j2]: Moved license header to the top. That is how it should be.

File: infrastructure/ansible/roles/traffic-router/defaults/main.yml [hunk 3]
```diff
@@ -177,4 +177,4 @@ tr_crontab:
   access_cleanup:
     schedule: '10 */4 * * *'
     user: root
-    job:  /bin/find /opt/traffic_router/var/log/access.log.* -mmin +$((60*48)) -exec rm -f {} \; > /dev/null 2>&1
+    job:  /bin/find {{tr_log_dir}}/access.log.* -mmin +$((60*48)) -exec rm -f {} \; > /dev/null 2>&1
```
Comment (jhg03a) [inline infrastructure/ansible/roles/traffic-router/defaults/main.yml]: When you use a var like that, you have to surround the line with double-quotes I believe so it knows interpolation should occur and not be a string literal.

**Review by jhg03a — CHANGES_REQUESTED** [review]

File: infrastructure/ansible/roles/traffic-router/defaults/main.yml (hunk 3 above)
Comment (jpappa200) [reply infrastructure/ansible/roles/traffic-router/defaults/main.yml]: yea missed that. it's fixed now.

**Review by jhg03a — APPROVED** [review]

**Pull request merged**